### PR TITLE
Change ButtonHandler to MenuBar

### LINKED ISSUE #112: feat: Keyboard Shortcuts
Traitsui allows the use of keyboard shortcuts thru "accelerators".  Everything that I see says the ButtonHandler should support them as well.  Try it out and/or alternatively try a menuBar.  https://docs.enthought.com/envisage/tasks_user_manual/menus.html

- Run "Ctrl + R"
- Quit "Ctrl + Q"
- Open Config "Ctrl + O"
- Save Config "Ctrl + S"
- Toggle Debug Mode "Ctrl + `"

```python
    handler=GlobalHandler(),
    menubar=MenuBar(
        Menu(
            Action(name="Load Config.", action="load_config_clicked", accelerator="Ctrl+O"),
            Action(name="Load Results", action="load_data_clicked"),
            Separator(),
            Action(name="Save Config.", action="save_config_clicked", accelerator="Ctrl+S"),
            Action(name="Save Results", action="save_data_clicked"),
            Separator(),
            # CloseAction, #! Unable to add due to https://github.com/enthought/traitsui/issues/1442
            name="&File",
        ),
        Menu(
            Action(
                name="Debug Mode",
                action="toggle_debug_clicked",
                accelerator="Ctrl+`",
                style="toggle",
                checked_when="debug == True",
            ),
            Action(
                name="Clear Reference Waveform(s)",
                action="clear_reference_clicked",
            ),
            name="&View",
        ),
        Menu(
            Action(name="Run", action="run_simulation_clicked", accelerator="Ctrl+R"),
            name="&Simulation"
        ),
        Menu(
            Action(name="Getting Started", action="getting_started_clicked"),
            Action(name="&About", action="show_about_clicked"),
            name="&Help",
        ),
```

### LINKED ISSUE #113: Ability to Clear Reference Waveforms
Right now "Load Results" loads a previous run as a reference waveform.  There is no mechanism to clear that waveform, you can only load another reference which will replace the previous reference.  You must close/reopen the GUI to clear it.  Add a menu action to clear the reference plots.

```python
    def clear_reference_clicked(self, info):
        """If the user has loaded previous "reference" data, clear that data from the plots."""

        # Remove the data from the ArrayPlotContainer
        for reference_plot in info.object.plotdata.list_data():
            if "ref" in reference_plot:
                try:
                    info.object.plotdata.del_data(reference_plot)
                except KeyError:
                    pass

        info.object.initialize_plots()
        info.object.update_results()
```

### PR DESCRIPTION
**Pull request by jdpatt**
This PR closes #112 and closes #113 by making two changes to the user interface.

### Adds
- It adds a button to be able to clear any waveforms that are loaded by the user instead of requiring them to close/re-open the GUI to clear them. 
- It adds keyboard short cuts so that you can run, open, save quickly in the GUI.  The full list of short cuts:
    - Run "Ctrl + R"
    - Quit "Ctrl + Q"
    - Open Config "Ctrl + O"
    - Save Config "Ctrl + S"
    - Save Config As "Ctrl + Shift + S"
    - Toggle Debug Mode "Ctrl + `"

### Changes
- The button handler at the bottom apparently doesn't support the shortcuts or accelerators as traitsui calls them.  Only the menuBar does so this moves the buttons from the bottom to the top but are still globally accessible.
- Removes the double newline in the debug console.
- Renames the last tab from "help" to "Info" since the help and about info are on the menubar.
- When the user clicks getting started, it opens a web browser to our wiki!

### Notes
-  OS X does special things when menus have certain names like about and quit are moved under the apple menu instead of under file and help like in windows.
-  CloseAction is broken in traitsui so we made our own.
- Traitsui doesn't honor order in the menuBar like it should.

OS X
Screen Shot 2022-12-08 at 9 05 54 PM
Windows 10
image

### CODE REVIEW

**Comment by jdpatt** [comment]
@capn-freako please try out the menu bar when you get a chance and some feedback before opening the PR.

File: src/pybert/gui/view.py [hunk 1]
```diff
@@ -122,15 +145,47 @@ def do_load_data(self, info):
...
+        """Close the app.
+
+        Can't use CloseAction until https://github.com/enthought/traitsui/issues/1442 is resolved.
+        """
+        # pylint: disable=no-self-use,unused-argument
+        sys.exit(0)
 
-# These are the "globally applicable" buttons referred to in pybert.py,
-# just above the button definitions (approx. line 580).
-run_sim = Action(name="Run", action="do_run_simulation")
```
Comment (capn-freako) [inline src/pybert/gui/view.py:128]: Is it possible to keep the Run/Stop buttons in the interface, or is it a case of mutual exclusivity between buttons and menus?

File: src/pybert/gui/view.py [hunk 2]
```diff
@@ -122,15 +145,47 @@ def do_load_data(self, info):
...
-stop_sim = Action(name="Stop", action="do_stop_simulation")
```
Comment (capn-freako) [inline src/pybert/gui/view.py:129]: What is our way of aborting a simulation, now?

File: src/pybert/gui/view.py [hunk 3]
```diff
@@ -209,7 +264,6 @@ def do_load_data(self, info):
                     Item(
                         name="thresh",
                         label="Pj Threshold (sigma)",
-                        tooltip="Threshold for identifying periodic jitter spectral elements. (sigma)",
```
Comment (capn-freako) [inline src/pybert/gui/view.py:212]: Why do you want to remove this tip?

File: src/pybert/gui/view.py [hunk 4]
```diff
@@ -812,25 +866,60 @@ def do_load_data(self, info):
...
+            label="Info",
+            id="info",
         ),
         layout="tabbed",
         springy=True,
         id="tabs",
     ),
     resizable=True,
     handler=MyHandler(),
-    buttons=[run_sim, save_cfg, load_cfg, save_data, load_data],
```
Comment (capn-freako) [inline src/pybert/gui/view.py:833]: It looks like we never actually had a "Stop" button in the interface.
Although, I see it defined above.
Was that something that you had started to add, before you switched to a menu based approach?

File: src/pybert/pybert.py [hunk 5]
```diff
@@ -615,11 +606,11 @@ def log(self, msg, alert=False, exception=None):
         """Log a message to the console and, optionally, to terminal and/or
         pop-up dialog."""
         _msg = msg.strip()
-        txt = f"\n[{datetime.now()}]: PyBERT: {_msg}\n"
+        txt = f"[{datetime.now()}]: PyBERT: {_msg}"
         if self.debug:
             ## In case PyBERT crashes, before we can read this in its `Console` tab:
             print(txt, flush=True)
-        self.console_log += txt
+        self.console_log += txt + "\n"
```
Comment (capn-freako) [inline src/pybert/pybert.py:613]: Wow! I really was in love w/ carriage returns, here, wasn't I? ;-)
Thanks for cleaning this up! :)

**Review by capn-freako — CHANGES_REQUESTED** [review]
Thanks for the sprint, David!
I really love what you've done here.
It's starting to look like a "real" application. ;-)
-db

File: src/pybert/gui/view.py (hunk 3 above)
Comment (jdpatt) [reply src/pybert/gui/view.py:212]: Wasn't intended. Reverted.

File: src/pybert/gui/view.py (hunk 4 above)
Comment (jdpatt) [reply src/pybert/gui/view.py:833]: I don't think a stop button has been exposed as long as I can remember but the code was always there.  I've never used or modified it.

Comment (jdpatt) [reply src/pybert/gui/view.py:833]: Back in 2018 was added but not to the buttons!  https://github.com/capn-freako/PyBERT/commit/ecb226cec08a87d34eab688adef00209670a8ca7

File: src/pybert/pybert.py (hunk 5 above)
Comment (jdpatt) [reply src/pybert/pybert.py:613]: Three when printing and two in the log. :) Can never have enough!

**Review by capn-freako — APPROVED** [review]
Thanks, David!

File: src/pybert/gui/view.py (hunk 4 above)
Comment (capn-freako) [reply src/pybert/gui/view.py:833]: D'oh!
Well, I guess if we've never missed it then we don't _really_ need it.
I'll add it to the _Issues_ list, as a non-gating feature request for later development.

File: src/pybert/gui/view.py (hunk 2 above)
Comment (capn-freako) [reply src/pybert/gui/view.py:129]: We don't currently have one.
Adding to _Issues_ as a low priority future development item...

File: src/pybert/gui/view.py (hunk 1 above)
Comment (capn-freako) [reply src/pybert/gui/view.py:128]: Yeah, and it chews up a fair amount of vertical screen real estate. :(
I already made a pass through the GUI layout (several years ago, now), specifically to make it usable in older Hx768 screen resolutions.

Let's stick w/ button-free.
Your new keyboard shortcuts are a better solution anyway, I think.

**Comment by capn-freako** [comment]
> @capn-freako please try out the menu bar when you get a chance.

Seems to be working completely as advertised on my end.
Thanks!
:)

**Pull request merged**